File: Labs/Lab 5/Lab5.py
def isValidBST(self, root):
        minVal = float('-inf')
        if root is None:
            return True
        if not self.isValidBST(root.left):
            return False
        if minVal >= root.val:
            return False
        minVal = root.val
        if not self.isValidBST(root.right):
            return False
        return True

File: Labs/Lab 5/Lab5.java
public boolean isValidBST(TreeNode root) {
        long minVal = Long.MIN_VALUE;
        if(root == null){
            return true;
        }
        if(!isValidBST(root.left)){
            return false;
        }
        if(minVal >= root.val){
            return false;
        }
        minVal = root.val;
        if(!isValidBST(root.right)){
            return false;
        }
        return true;
}
